src/mumble/GlobalShortcut_win.cpp

```diff
@@ -130,576 +52,356 @@ GlobalShortcutWin::~GlobalShortcutWin() {
-
-#ifndef QT_NO_DEBUG
-	static int safety = 0;
+void GlobalShortcutWin::on_mouseMessage(const uint16_t flags, const uint16_t data) {
+	uint8_t button;
+	bool down;
 
-	if ((++safety < 100) && (nCode >= 0)) {
-#else
-	if (nCode >= 0) {
-#endif
-		DWORD scancode = key->scanCode;
-		DWORD vkcode   = key->vkCode;
-		bool extended  = !!(key->flags & LLKHF_EXTENDED);
-		bool up        = !!(key->flags & LLKHF_UP);
-		bool suppress  = handleKeyboardMessage(scancode, vkcode, extended, !up);
-		if (suppress) {
-			return 1;
-		}
+	switch (flags) {
+		case RI_MOUSE_BUTTON_1_DOWN:
+			// Left button pressed
+			button = 1;
+			down   = true;
+			break;
+		case RI_MOUSE_BUTTON_1_UP:
+			// Left button released
+			button = 1;
+			down   = false;
+			break;
+		case RI_MOUSE_BUTTON_2_DOWN:
+			// Right button pressed
+			button = 2;
+			down   = true;
+			break;
+		case RI_MOUSE_BUTTON_2_UP:
+			// Right button released
+			button = 2;
+			down   = false;
+			break;
+		case RI_MOUSE_BUTTON_3_DOWN:
+			// Middle button pressed
+			button = 3;
+			down   = true;
+			break;
+		case RI_MOUSE_BUTTON_3_UP:
+			// Middle button released
+			button = 3;
+			down   = false;
+			break;
+		case RI_MOUSE_BUTTON_4_DOWN:
+			// Side button 1 pressed
+			button = 4;
+			down   = true;
+			break;
+		case RI_MOUSE_BUTTON_4_UP:
+			// Side button 1 released
+			button = 4;
+			down   = false;
+			break;
+		case RI_MOUSE_BUTTON_5_DOWN:
+			// Side button 2 pressed
+			button = 5;
+			down   = true;
+			break;
+		case RI_MOUSE_BUTTON_5_UP:
+			// Side button 2 released
+			button = 5;
+			down   = false;
+			break;
+		default:
+			return;
 	}
-	return CallNextHookEx(gsw->hhKeyboard, nCode, wParam, lParam);
-}
 
-LRESULT CALLBACK GlobalShortcutWin::HookMouse(int nCode, WPARAM wParam, LPARAM lParam) {
-	GlobalShortcutWin *gsw = static_cast< GlobalShortcutWin * >(engine);
-	MSLLHOOKSTRUCT *mouse  = reinterpret_cast< MSLLHOOKSTRUCT * >(lParam);
-
-	if (nCode >= 0) {
-		bool suppress = false;
-		UINT msg      = wParam;
-		// Convert the hooked Windows mouse message into a DirectInput
-		// button index, and store the pressed state of the button.
-		bool down        = false;
-		unsigned int btn = 0;
-		switch (msg) {
-			case WM_LBUTTONDOWN:
-				down = true;
-			case WM_LBUTTONUP:
-				btn = 3;
-				break;
-			case WM_RBUTTONDOWN:
-				down = true;
-			case WM_RBUTTONUP:
-				btn = 4;
-				break;
-			case WM_MBUTTONDOWN:
-				down = true;
-			case WM_MBUTTONUP:
-				btn = 5;
-				break;
-			case WM_XBUTTONDOWN:
-				down = true;
-			case WM_XBUTTONUP:
-				btn = 5 + (mouse->mouseData >> 16);
-			default:
-				break;
-		}
-		if (btn > 0) {
-			suppress = handleMouseMessage(btn, down);
-			if (suppress) {
-				return 1;
-			}
-		}
-	}
-	return CallNextHookEx(gsw->hhMouse, nCode, wParam, lParam);
+	const QList< QVariant > ql{ QVariant::fromValue(DeviceTypes::MOUSE), button };
+	handleButton(ql, down) && button != 1;
```

Comment:
What is this construct supposed to do? It seems like the boolean created here is simply discarded? :eyes:

Reply:
You're right, it's a leftover from the previous code.